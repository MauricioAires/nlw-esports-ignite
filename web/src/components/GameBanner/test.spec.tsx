import { render, screen } from '@testing-library/react';
import { GameBanner } from '.';

const sut = () =>
  render(
    <GameBanner adsCount={1} bannerUrl="/img.jpg" title="game banner title" />
  );

describe('<GameBanner />', () => {
  it('should render correctly', () => {
    const { container } = sut();

    expect(screen.getByRole('img')).toHaveAttribute('src', '/img.jpg');
    expect(screen.getByRole('img')).toHaveAttribute('alt', 'game banner title');
    expect(
      screen.getByRole('heading', {
        name: /game banner title/i,
      })
    ).toBeInTheDocument();
    expect(screen.getByText(/1 an√∫ncio\(s\)/)).toBeInTheDocument();

    expect(container.firstElementChild).toMatchSnapshot();
  });
});
