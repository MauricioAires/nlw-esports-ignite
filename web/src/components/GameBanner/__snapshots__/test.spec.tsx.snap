// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameBanner /> should render correctly 1`] = `
<a
  class="relative rounded-lg overflow-hidden"
  href=""
>
  <img
    alt="game banner title"
    src="/img.jpg"
  />
  <div
    class="w-full pt-16 pb-4 px-4 bg-game-gradient absolute bottom-0 left-0 right-0"
  >
    <h2
      class="font-bold text-white block"
    >
      game banner title
    </h2>
    <span
      class="text-zinc-300 block "
    >
      1
       an√∫ncio(s)
    </span>
  </div>
</a>
`;
